src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -547,36 +547,53 @@ private void AddTypeKind(INamedTypeSymbol symbol)
                 }
                 else
                 {
-                    var kindKeyword = GetKindKeyword(symbol.TypeKind);
-                    if (kindKeyword != SyntaxKind.None)
+                    switch (symbol.TypeKind)
                     {
-                        AddKeyword(kindKeyword);
-                        AddSpace();
+                        case TypeKind.Module:
+                        case TypeKind.Class:
+                            AddKeyword(SyntaxKind.ClassKeyword);
+                            AddSpace();
+                            break;
+
+                        case TypeKind.Enum:
+                            AddKeyword(SyntaxKind.EnumKeyword);
+                            AddSpace();
+                            break;
+
+                        case TypeKind.Delegate:
+                            AddKeyword(SyntaxKind.DelegateKeyword);
+                            AddSpace();
+                            break;
+
+                        case TypeKind.Interface:
+                            AddKeyword(SyntaxKind.InterfaceKeyword);
+                            AddSpace();
+                            break;
+
+                        case TypeKind.Struct:
+                            if (symbol is NamedTypeSymbol csharpType)
+                            {
+                                if (csharpType.IsReadOnly)
```

Comment:
Is it ok to add “readonly” unconditionally? 
It is kind of verbose and in most cases does not affect the use of the type.  

Perhaps it should depend on some symbol format flag?

We can follow up on this in a different change though.

Reply:
I'm impartial to this, but if we added now, it would be a break to hide it later behind a flag. Thoughts?
@gafter for API change (should we add a flag?)

Reply:
We have `SymbolDisplay*Options` for most qualifiers. It seems like `readonly ref` should be controlled by an option as well.

Reply:
And my sense is that ref and readonly should be able to switch separately. But I also see that too many knobs could be unwieldy.